File: isegoria/src/main/java/com/eulersbridge/isegoria/feed/FeedViewModel.java
package com.eulersbridge.isegoria.feed;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.support.annotation.NonNull;

import com.eulersbridge.isegoria.IsegoriaApp;

@SuppressWarnings("WeakerAccess")
public class FeedViewModel extends AndroidViewModel {

    public FeedViewModel(@NonNull Application application) {
        super(application);
    }

    void showFriends() {
        IsegoriaApp app = getApplication();
        app.friendsVisible.setValue(true);
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/feed/FeedViewModel.kt
package com.eulersbridge.isegoria.feed

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import com.eulersbridge.isegoria.AppRouter

class FeedViewModel(application: Application) : AndroidViewModel(application) {

    internal fun showFriends() {
        val app = getApplication<Application>() as? AppRouter
        app!!.setFriendsScreenVisible(true)
    }
}
